go/mysql/mysql56_gtid_set.go

```diff
@@ -435,6 +435,128 @@ func (set Mysql56GTIDSet) SIDBlock() []byte {
 	return buf.Bytes()
 }
 
+func (set Mysql56GTIDSet) Difference(other Mysql56GTIDSet) Mysql56GTIDSet {
+	if other == nil {
+		return set
+	}
+
+	// Make a fresh, empty set to hold the new value.
+	// This function is not supposed to modify the original set.
+	differenceSet := make(Mysql56GTIDSet)
```

Comment:
if this is something that is used in the hot path, and will contain more than just a few items, you might want to consider using make with a given size, to avoid resizing hash maps

Reply:
I was thinking about that, but we don't know yet what the size of the `differenceSet` will be. Are you thinking that we should just make it the same size as the receiver to cover all potential SIDs we might find that have valid diffs?